shared/api_archive/archive.py

```diff
@@ -64,7 +64,10 @@ def __init__(self, repository, ttl=None):
         self.ttl = ttl or int(get_config("services", "minio", "ttl", default=self.ttl))
 
         self.storage = StorageService()
-        self.storage_hash = self.get_archive_hash(repository)
+        if repository:
+            self.storage_hash = self.get_archive_hash(repository)
+        else:
+            self.storage_hash = None
```

Comment:
Making ArchiveService dependent on a repository was complicating my flare cleanup task.
It seems like repository is only needed to make the hash for reading and writing to the archive - you don't need it to delete.
To make my flare cleanup task much lighter, I removed the requirement for a repository in order to use ArchiveService. I added checks to the ArchiveStorage functions where the hash is used, so that you can't do something like write to storage without a hash.